parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -132,7 +135,9 @@ private void ensureWriterInitialized() {
   @Override
   public void add(T value) {
     recordCount += 1;
+    long sizeBeforeWrite = writeStore.getBufferedSize();
```

Comment:
We are now calling getBufferedSize for every value, this seems like a big perf difference? Can we run the write benchmarks with this on and off?